cli/azd/cmd/vs_server.go

```diff
@@ -61,19 +71,33 @@ func newVsServerAction(rootContainer *ioc.NestedContainer, flags *vsServerFlags)
 func (s *vsServerAction) Run(ctx context.Context) (*actions.ActionResult, error) {
 	listener, err := net.Listen("tcp", fmt.Sprintf("127.0.0.1:%d", s.flags.port))
 	if err != nil {
-		panic(err)
+		return nil, err
+	}
+
+	cert, derBytes, err := generateCertificate()
+	if err != nil {
+		return nil, err
 	}
 
+	config := &tls.Config{
+		MinVersion:   tls.VersionTLS12,
+		NextProtos:   []string{"http/1.1"},
+		Certificates: []tls.Certificate{cert},
+	}
+
+	tlsListener := tls.NewListener(listener, config)
```

Comment:
@weikanglim - I considered adding support for some `AZD_DEBUG` env var like `AZD_DEBUG_SERVER_ALLOW_INSECURE` and when that was true according to `strconv.ParseBool` not doing this TLS stuff.  I'm not sure if that switch would be helpful to have as a point in time thing as we land this change to allow folks to run newer AZD's in a way that older clients (like VS) can still connect to.

Happy to add this is you think it is worthwhile - I assume that the the landing of the "TLS Everywhere" stuff is going to be a little rocky because we have to coordinate with VS - so if you think that would help, I am happy to do it.

Reply:
After some internal discussion with our VS partners, I've changed this such that you opt into this behavior right now by calling `azd vs-server` with `--use-tls`.